crates/worker/src/partition/services/non_deterministic/remote_context.rs

```diff
@@ -479,16 +459,33 @@ impl<'a, State: StateReader> InvocationContext<'a, State> {
         &mut self,
         operation_id: &str,
         retention_period_sec: u32,
-        entry_index: EntryIndex,
-        result: ResponseResult,
+        virtual_journal_length: EntryIndex,
     ) -> Result<(), InvocationError> {
         let expiry_time = SystemTime::now() + Duration::from_secs(retention_period_sec as u64);
+        let journal_service_id = self.journal_service_id();
+
+        // Load the result from the journal last entry
+        let output_stream_entry_index = virtual_journal_length - 1;
+        let entry = self
+            .read_journal_entry(&journal_service_id, output_stream_entry_index)
```

Comment:
Did you introduce the journal entry caching in `StateTransitions` to be able to read an `OutputStreamEntry` which hasn't been written to RocksDB yet here?

Reply:
Yes.